projects/ngx-drag-to-select/src/lib/select-container.component.ts

```diff
@@ -410,21 +421,61 @@ export class SelectContainerComponent implements AfterViewInit, OnDestroy {
       this.renderer.addClass(document.body, NO_SELECT_CLASS);
     }
 
+    if (this.shortcuts.removeFromSelection(event)) {
+      return;
+    }
+
     const mousePoint = getMousePosition(event);
+    const [currentIndex, clickedItem] = this._getClosestSelectItem(event);
+
+    let [startIndex, endIndex] = this._lastRange;
+
+    if (!this.shortcuts.extendedSelectionShortcut(event) && currentIndex > -1) {
+      const lastRangeStart = this._selectableItems[this._lastStartIndex];
+
+      if (lastRangeStart) {
+        lastRangeStart.toggleRangeStart();
+      }
+
+      this._lastStartIndex = currentIndex;
+      clickedItem.toggleRangeStart();
+    }
+
+    if (!this.shortcuts.extendedSelectionShortcut(event) && currentIndex === -1) {
+      if (this._lastStartIndex >= 0) {
+        const lastStart = this._selectableItems[this._lastStartIndex];
+        lastStart.toggleRangeStart();
+      }
+
+      this._lastStartIndex = -1;
+    }
```

Comment:
```ts
if (!this.shortcuts.extendedSelectionShortcut(event)) {
     const lastRangeStart = this._selectableItems[this._lastStartIndex];
      if (lastRangeStart) {
        lastRangeStart.toggleRangeStart();
      }

    this._lastStartIndex = currentIndex;
    if(this._lastStartIndex  !== 1) { 
      clickedItem.toggleRangeStart();
   }
} 
```

OR via a setter?
The below is pseudo-code
```ts
private _lastStartIndex set(index: number | undefined) {
   if(previousLastStartIndex) {
     items[previousLastStartIndex].toggle()
   }


  items[index].toggle();
  previousLastStartIndex = index;
}

```

Reply:
Yep, thanks! I have just pushed my changes. It refactors this particular piece as well. Nice spotting 👍